sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/MetricName.java

```diff
@@ -0,0 +1,33 @@
+package org.apache.beam.sdk.metrics;
+
+import com.google.auto.value.AutoValue;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.annotations.Experimental.Kind;
+
+/**
+ * The name of a metric.
+ */
+@Experimental(Kind.METRICS)
+@AutoValue
+public abstract class MetricName {
+
+  /** The namespace that declared this metric. */
+  public abstract String getNamespace();
+
+  /**
+   * The name of this metric. Should only be null in {@link MetricFilter MetricFilters} being used
```

Comment:
I thought about that, but what do we do if they are both set? Do we intersect them so that only metrics that are listed in both the set of namespace/name pairs you've requested AND the list of namespaces you've requested show up? Or do we treat it as OR (even though other filters are treated as AND).

Maybe the right solution would be to have a separate type for "MetricNameFilter" that makes the name optional, and use that in filters?